diff --git a/build.gradle b/build.gradle
--- a/build.gradle	(revision fc40aa9d88e9457957bdf3f8cec9698846828cd3)
+++ b/build.gradle	(date 1631009569915)
@@ -257,6 +257,9 @@
 		setupRepositories(repositories)
 	}

+	// Required as moduleDependencies modifies the pom
+	loom.disableDeprecatedPomGeneration(publishing.publications.mavenJava)
+
 	javadoc.enabled = false
 }

@@ -296,6 +299,8 @@
 	setupRepositories(repositories)
 }

+loom.disableDeprecatedPomGeneration(publishing.publications.mavenJava)
+
 void setupRepositories(RepositoryHandler repositories) {
 	//repositories.mavenLocal() // uncomment for testing
 	def ENV = System.getenv()
