diff --git a/build.gradle b/build.gradle
--- a/build.gradle	(revision f84dc5662589fd56ac4b36a4b94920a15b1da29d)
+++ b/build.gradle	(date 1749475915352)
@@ -9,7 +9,7 @@
 }

 def branchProvider = providers.of(GitBranchValueSource.class) {}
-version = project.version + "+" + (providers.environmentVariable("CI").present ? branchProvider.get().replace("/", "_") : "local")
+version = "999.0.0"
 logger.lifecycle("Building Fabric: " + version)

 def metaProjects = [
@@ -33,22 +33,8 @@
 import java.net.http.HttpRequest
 import java.net.http.HttpResponse

-def getSubprojectVersion(Project project) {
-	// Get the version from the gradle.properties file
-	def version = project.properties["${project.name}-version"]
-
-	if (!version) {
-		throw new NullPointerException("Could not find version for " + project.name)
-	}
-
-	if (!project.providers.environmentVariable("CI").present) {
-		return version + "+local"
-	}
-
-	def hashProvider = project.providers.of(CommitHashValueSource.class) {
-		parameters.directory = project.name
-	}
-	return version + "+" + hashProvider.get().substring(0, 8) + DigestUtils.sha256Hex(project.rootProject.minecraft_version).substring(0, 2)
+def getSubprojectVersion(project) {
+	return "999.0.0"
 }

 def moduleDependencies(project, List<String> depNames) {
