diff --git a/build.gradle b/build.gradle
--- a/build.gradle	(revision 14d319c0729baf781e171e3c9f845fda55670f1b)
+++ b/build.gradle	(date 1688330748664)
@@ -37,17 +37,7 @@
 		throw new NullPointerException("Could not find version for " + project.name)
 	}
 
-	if (grgit == null) {
-		return version + "+nogit"
-	}
-
-	def latestCommits = grgit.log(paths: [project.name], maxCommits: 1)
-
-	if (latestCommits.isEmpty()) {
-		return version + "+uncommited"
-	}
-
-	return version + "+" + latestCommits.get(0).id.substring(0, 8) + DigestUtils.sha256Hex(project.rootProject.minecraft_version).substring(0, 2)
+	return version
 }
 
 def getBranch() {
