diff --git a/gradle/validate-annotations.gradle b/gradle/validate-annotations.gradle
--- a/gradle/validate-annotations.gradle	(revision f84dc5662589fd56ac4b36a4b94920a15b1da29d)
+++ b/gradle/validate-annotations.gradle	(date 1779437055038)
@@ -13,7 +13,7 @@

 tasks.check.dependsOn "validateAnnotations"

-class ValidateAnnotations extends SourceTask {
+abstract class ValidateAnnotations extends SourceTask {
 	private static final def API_STATUS_INTERNAL = ~/@ApiStatus\.Internal/
 	private static final def ENVIRONMENT = ~/@Environment/

Index: gradle.properties
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/gradle.properties b/gradle.properties
--- a/gradle.properties	(revision f84dc5662589fd56ac4b36a4b94920a15b1da29d)
+++ b/gradle.properties	(date 1779491776054)
@@ -4,7 +4,7 @@
 version=0.126.0
 minecraft_version=1.21.6-pre3
 yarn_version=+build.1
-loader_version=0.16.13
+loader_version=0.19.2
 installer_version=1.0.1

 prerelease=true
Index: build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/build.gradle b/build.gradle
--- a/build.gradle	(revision f84dc5662589fd56ac4b36a4b94920a15b1da29d)
+++ b/build.gradle	(date 1779491776048)
@@ -2,14 +2,14 @@
 	id "java-library"
 	id "eclipse"
 	id "maven-publish"
-	id "fabric-loom" version "1.10.1" apply false
+	id "fabric-loom" apply false
 	id "com.diffplug.spotless" version "7.0.2"
 	id "me.modmuss50.remotesign" version "0.5.0" apply false
 	id "me.modmuss50.mod-publish-plugin" version "0.8.4"
 }

 def branchProvider = providers.of(GitBranchValueSource.class) {}
-version = project.version + "+" + (providers.environmentVariable("CI").present ? branchProvider.get().replace("/", "_") : "local")
+version = "999.0.0"
 logger.lifecycle("Building Fabric: " + version)

 def metaProjects = [
@@ -33,22 +33,8 @@
 import java.net.http.HttpRequest
 import java.net.http.HttpResponse

-def getSubprojectVersion(Project project) {
-	// Get the version from the gradle.properties file
-	def version = project.properties["${project.name}-version"]
-
-	if (!version) {
-		throw new NullPointerException("Could not find version for " + project.name)
-	}
-
-	if (!project.providers.environmentVariable("CI").present) {
-		return version + "+local"
-	}
-
-	def hashProvider = project.providers.of(CommitHashValueSource.class) {
-		parameters.directory = project.name
-	}
-	return version + "+" + hashProvider.get().substring(0, 8) + DigestUtils.sha256Hex(project.rootProject.minecraft_version).substring(0, 2)
+def getSubprojectVersion(project) {
+	return "999.0.0"
 }

 def moduleDependencies(project, List<String> depNames) {
@@ -355,7 +341,6 @@

 // Apply auxiliary buildscripts to submodules
 // This must be done after all plugins are applied to subprojects
-apply from: "gradle/module-validation.gradle"
 apply from: "gradle/module-versioning.gradle"

 loom {
@@ -530,15 +515,15 @@
 		// Make all modules depend on the gametest api (and thus res loader) to try and promote its usage.
 		if (project.name != "fabric-gametest-api-v1") {
 			testmodImplementation project(path: ':fabric-gametest-api-v1', configuration: 'namedElements')
-			testmodClientImplementation project(":fabric-gametest-api-v1").sourceSets.client.output
+			testmodClientImplementation this.project(":fabric-gametest-api-v1").sourceSets.client.output
 			testmodImplementation project(path: ':fabric-resource-loader-v0', configuration: 'namedElements')
-			testmodClientImplementation project(":fabric-resource-loader-v0").sourceSets.client.output
+			testmodClientImplementation this.project(":fabric-resource-loader-v0").sourceSets.client.output
 		}

 		// Make all testmods run with registry-sync-v0 as it is required to register new objects.
 		if (project.name != "fabric-registry-sync-v0") {
 			testmodRuntimeOnly project(path: ':fabric-registry-sync-v0', configuration: 'namedElements')
-			testmodClientImplementation project(":fabric-registry-sync-v0").sourceSets.client.output
+			testmodClientImplementation this.project(":fabric-registry-sync-v0").sourceSets.client.output
 		}
 	}

@@ -694,10 +679,10 @@
 			}

 			api project(path: "${it.path}", configuration: "namedElements")
-			clientImplementation project("${it.path}:").sourceSets.client.output
+			clientImplementation this.project("${it.path}:").sourceSets.client.output

-			testmodImplementation project("${it.path}:").sourceSets.testmod.output
-			testmodClientImplementation project("${it.path}:").sourceSets.testmodClient.output
+			testmodImplementation this.project("${it.path}:").sourceSets.testmod.output
+			testmodClientImplementation this.project("${it.path}:").sourceSets.testmodClient.output
 		}
 	}
 }
@@ -771,7 +756,7 @@
 tasks.register('checkVersion') {
 	doFirst {
 		def xml = new URL("https://maven.fabricmc.net/net/fabricmc/fabric-api/fabric-api/maven-metadata.xml").text
-		def metadata = new XmlSlurper().parseText(xml)
+		def metadata = new groovy.xml.XmlSlurper().parseText(xml)
 		def versions = metadata.versioning.versions.version*.text();
 		if (versions.contains(version)) {
 			throw new RuntimeException("${version} has already been released!")
