diff --git a/build.gradle b/build.gradle
--- a/build.gradle	(revision 41bc64cd617f03d49ecc4a4f7788cb65d465415c)
+++ b/build.gradle	(date 1718312645477)
@@ -13,7 +13,7 @@

 def ENV = System.getenv()

-version = project.version + "+" + (ENV.GITHUB_RUN_NUMBER ? "" : "local-") + getBranch()
+version = "999.0.0"
 logger.lifecycle("Building Fabric: " + version)

 def metaProjects = [
@@ -34,24 +34,7 @@
 import org.apache.commons.codec.digest.DigestUtils

 def getSubprojectVersion(project) {
-	// Get the version from the gradle.properties file
-	def version = project.properties["${project.name}-version"]
-
-	if (!version) {
-		throw new NullPointerException("Could not find version for " + project.name)
-	}
-
-	if (grgit == null) {
-		return version + "+nogit"
-	}
-
-	def latestCommits = grgit.log(paths: [project.name], maxCommits: 1)
-
-	if (latestCommits.isEmpty()) {
-		return version + "+uncommited"
-	}
-
-	return version + "+" + latestCommits.get(0).id.substring(0, 8) + DigestUtils.sha256Hex(project.rootProject.minecraft_version).substring(0, 2)
+	return "999.0.0"
 }
 
 def getBranch() {
