diff --git a/build.gradle b/build.gradle
--- a/build.gradle	(revision 5cda40cfe1ddb507dd509c89117b0801a6ab79d9)
+++ b/build.gradle	(date 1659386544411)
@@ -33,17 +33,7 @@
 		throw new NullPointerException("Could not find version for " + project.name)
 	}

-	if (grgit == null) {
-		return version + "+nogit"
-	}
-
-	def latestCommits = grgit.log(paths: [project.name], maxCommits: 1)
-
-	if (latestCommits.isEmpty()) {
-		return version + "+uncommited"
-	}
-
-	return version + "+" + latestCommits.get(0).id.substring(0, 8) + DigestUtils.sha256Hex(project.rootProject.minecraft_version).substring(0, 2)
+	return version
 }

 def getBranch() {
