diff --git a/build.gradle b/build.gradle
--- a/build.gradle	(revision dfa883a2e9348fc4a877252be6dbafc264f19006)
+++ b/build.gradle	(date 1779491895630)
@@ -2,14 +2,14 @@
 	id "java-library"
 	id "eclipse"
 	id "maven-publish"
-	id "net.fabricmc.fabric-loom" version "1.16.2" apply false
+	id "net.fabricmc.fabric-loom" apply false
 	id "com.diffplug.spotless" version "8.5.1"
 	id "me.modmuss50.remotesign" version "0.5.0" apply false
 	id "me.modmuss50.mod-publish-plugin" version "2.0.0-beta.1"
 }

 def branchProvider = providers.of(GitBranchValueSource.class) {}
-version = project.version + "+" + (providers.environmentVariable("CI").present ? branchProvider.get().replace("/", "_") : "local")
+version = "999.0.0"
 logger.lifecycle("Building Fabric: " + version)

 def metaProjects = [
@@ -37,22 +37,8 @@
 import java.net.http.HttpRequest
 import java.net.http.HttpResponse

-def getSubprojectVersion(Project project) {
-	// Get the version from the gradle.properties file
-	def version = project.properties["${project.name}-version"]
-
-	if (!version) {
-		throw new NullPointerException("Could not find version for " + project.name)
-	}
-
-	if (!project.providers.environmentVariable("CI").present) {
-		return version + "+local"
-	}
-
-	def hashProvider = project.providers.of(CommitHashValueSource.class) {
-		parameters.directory = project.name
-	}
-	return version + "+" + hashProvider.get().substring(0, 8) + DigestUtils.sha256Hex(project.rootProject.minecraft_version).substring(0, 2)
+def getSubprojectVersion(project) {
+	return "999.0.0"
 }

 def moduleDependencies(project, List<String> depNames) {
