diff --git a/build.gradle b/build.gradle
--- a/build.gradle	(revision 71b634e5b7845296b11be3fa6545f4fbfacc017f)
+++ b/build.gradle	(date 1638654919842)
@@ -31,17 +31,7 @@
 		throw new NullPointerException("Could not find version for " + project.name)
 	}

-	if (grgit == null) {
-		return version + "+nogit"
-	}
-
-	def latestCommits = grgit.log(paths: [project.name], maxCommits: 1)
-
-	if (latestCommits.isEmpty()) {
-		return version + "+uncommited"
-	}
-
-	return version + "+" + latestCommits.get(0).id.substring(0, 8) + DigestUtils.sha256Hex(project.rootProject.minecraft_version).substring(0, 2)
+	return version
 }

 def getBranch() {
